main.go

```diff
@@ -109,6 +115,52 @@ func newRootCommand() *cobra.Command {
 	return cmd
 }
 
+func sleepUntilTERM() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, os.Interrupt, syscall.SIGTERM)
+
+	for {
+		select {
+		case sig := <-sigs:
```

Comment:
I think we do not need select here, we should just have a blocking channel here
```
if sig := <-sigs; sig == syscall.SIGTERM {
    signal.Stop(sigs)
    klog.Info("SIGTERM caught")
    break
}

Reply:
Well.. yeah, that would work as well. Isn't pretty much the same though? Easy to change anyway..